### PR TITLE
[libc][test] adds LINK_LIBRARIES pthread to pthreads test cmake

### LINKED ISSUE #89261: [libc][test] link error running pthreads tests on linux, x86_64
getting the follow error running `ninja check-libc` on latest. think some `LINK_LIBRARIES` just needs to be added to the pthreads test cmake file? will make a PR for this

```
/usr/bin/clang++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -g  projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o -o projects/libc/test/src/pthread/libc.test.src.pthread.pthread_condattr_test.__build__  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_destroy.__internal__.dir/./pthread_condattr_destroy.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_getclock.__internal__.dir/./pthread_condattr_getclock.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_getpshared.__internal__.dir/./pthread_condattr_getpshared.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_init.__internal__.dir/./pthread_condattr_init.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_setclock.__internal__.dir/./pthread_condattr_setclock.cpp.o  projects/libc/src/pthread/CMakeFiles/libc.src.pthread.pthread_condattr_setpshared.__internal__.dir/./pthread_condattr_setpshared.cpp.o  projects/libc/src/__support/StringUtil/CMakeFiles/libc.src.__support.StringUtil.error_to_string.dir/./error_to_string.cpp.o  projects/libc/src/__support/OSUtil/linux/CMakeFiles/libc.src.__support.OSUtil.linux.linux_util.dir/./quick_exit.cpp.o  projects/libc/src/errno/CMakeFiles/libc.src.errno.errno.__internal__.dir/./libc_errno.cpp.o  lib/libLibcDeathTestExecutors.unit.a  lib/libLibcTest.unit.a
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_GetDefaultValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:30: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:32: undefined reference to `pthread_condattr_getpshared'
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_SetGoodValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:46: undefined reference to `pthread_condattr_setclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:47: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:49: undefined reference to `pthread_condattr_setpshared'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:50: undefined reference to `pthread_condattr_getpshared'
/usr/bin/ld: projects/libc/test/src/pthread/CMakeFiles/libc.test.src.pthread.pthread_condattr_test.__build__.dir/pthread_condattr_test.cpp.o: in function `LlvmLibcPThreadCondAttrTest_SetBadValues::Run()':
/home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:64: undefined reference to `pthread_condattr_setclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:65: undefined reference to `pthread_condattr_getclock'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:67: undefined reference to `pthread_condattr_setpshared'
/usr/bin/ld: /home/seclab/repos/llvm-project/libc/test/src/pthread/pthread_condattr_test.cpp:68: undefined reference to `pthread_condattr_getpshared'
```

### PR DESCRIPTION
**Pull request by ashfordium**
Fixes #89261.

cc @nickdesaulniers 

fixes this for me. seems like others might not have the same issue, so lmk if i'm just building this wrong.

### CODE REVIEW

File: libc/test/src/pthread/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +40,8 @@ add_libc_unittest(
     libc.src.pthread.pthread_mutexattr_setpshared
     libc.src.pthread.pthread_mutexattr_setrobust
     libc.src.pthread.pthread_mutexattr_settype
+  LINK_LIBRARIES
+    -pthread
```
Comment (nickdesaulniers) [inline libc/test/src/pthread/CMakeLists.txt:44]: Your report in #89261 only specifies issues for libc/test/src/pthread/pthread_condattr_test.cpp. Are the changes to pthread_mutexattr_test or pthread_attr_test necessary?

**Review by nickdesaulniers — CHANGES_REQUESTED** [review]

File: libc/test/src/pthread/CMakeLists.txt (hunk 1 above)
Comment (Flandini) [reply libc/test/src/pthread/CMakeLists.txt]: No, they are not. Removed.

**Comment by michaelrj-google** [comment]
This shouldn't be necessary, since `-lpthread` just tells the compiler to link in the system's pthread library. If this test is actually using our library it shouldn't have any effect since we don't build a separate `libpthread`. Maybe try cleaning your build directory (with `rm -r build/projects/libc`) and rebuilding?

**Comment by nickdesaulniers** [comment]
Do you have eefee382186005d3662958e076c8e61e286ea1ab on your branch?

**Comment by nickdesaulniers** [comment]
Please let us know if https://github.com/llvm/llvm-project/pull/89308 helps resolve this issue for you? Sorry for my mistake creating a problem!

**Comment by Flandini** [comment]
> Please let us know if #89308 helps resolve this issue for you? Sorry for my mistake creating a problem!

Yup, that fixed it, thanks!

**Pull request closed without merging**